### PR TITLE
Update edt to 0.25

### PR DESCRIPTION
**Pull request by bakridemos**
Description: needed adjustments to make core compatible with the new Edt version (from 0.24.42 to 0.25)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
         "ext-libxml": "*",
         "ext-mbstring": "*",
         "ext-openssl": "*",
+        "demos-europe/demosplan-addon": "dev-upgrade-edt-to-0.25",
```
Comment (mussbach) [inline composer.json:19]: needs to be changed before merging

File: demosplan/DemosPlanCoreBundle/Command/FrontendIntegratorCommand.php [hunk 2]
```diff
@@ -51,9 +55,12 @@ class FrontendIntegratorCommand extends CoreCommand
     public function __construct(
         private readonly CurrentUserInterface $currentUser,
         private readonly JsApiResourceDefinitionBuilder $resourceDefinitionBuilder,
-        private readonly OpenAPISchemaGenerator $apiDocumentationGenerator,
+        // private readonly OpenAPISchemaGenerator $apiDocumentationGenerator,
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Command/FrontendIntegratorCommand.php:58]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Command/FrontendIntegratorCommand.php [hunk 3]
```diff
@@ -142,7 +151,18 @@ private function getOpenApiSpec(): OpenApi
         $allPermissions = Yaml::parseFile(DemosPlanPath::getConfigPath(Permissions::PERMISSIONS_YML));
         $this->currentUser->getPermissions()->enablePermissions(array_keys($allPermissions));
 
-        $openApiSpec = $this->apiDocumentationGenerator->getOpenAPISpecification();
+        // $openApiSpec = $this->apiDocumentationGenerator->getOpenAPISpecification();
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Command/FrontendIntegratorCommand.php:154]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Controller/Base/APIDocumentationController.php [hunk 4]
```diff
@@ -35,15 +38,28 @@ public function indexAction(): Response
...
     #[Route(path: '/api/openapi.json', methods: ['GET', 'HEAD'], options: ['expose' => true], name: 'dplan_api_openapi_json')]
-    public function jsonAction(OpenAPISchemaGenerator $apiDocumentationGenerator): Response
+    public function jsonAction(Manager $manager, RouterInterface $router, TranslatorInterface $translator): Response
     {
         if ('dev' !== $this->globalConfig->getKernelEnvironment()) {
             return $this->redirectToRoute('core_home');
         }
 
-        $openApi = $apiDocumentationGenerator->getOpenAPISpecification();
+        // $openApi = $apiDocumentationGenerator->getOpenAPISpecification();
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Controller/Base/APIDocumentationController.php:51]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/ApiRequest/JsonApiEsService.php [hunk 5]
```diff
@@ -168,10 +168,13 @@ public function getEsFilteredObjects(
...
         // prepare fetching real entities from Doctrine
         $esIds = array_column($esResultArrays, 'id');
-        $condition = $this->conditionFactory->propertyHasAnyOfValues(
-            $esIds,
-            $resourceType->id
-        );
+        $condition = [] === $esIds
+            ? $this->conditionFactory->false()
+            : $this->conditionFactory->propertyHasAnyOfValues($esIds, $resourceType->id);
+        /*  $condition = $this->conditionFactory->propertyHasAnyOfValues(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/ApiRequest/JsonApiEsService.php:174]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/AssessmentTable/AbstractRpcStatementBulkAction.php [hunk 6]
```diff
@@ -166,10 +166,13 @@ private function generateMethodResult(object $rpcRequest): object
      */
     private function loadRequestedStatements(array $statementIds, string $procedureId): array
     {
-        $idCondition = $this->conditionFactory->propertyHasAnyOfValues(
+        $idCondition = [] === $statementIds
+            ? $this->conditionFactory->false()
+            : $this->conditionFactory->propertyHasAnyOfValues($statementIds, $this->statementResourceType->id);
+        /*$idCondition = $this->conditionFactory->propertyHasAnyOfValues(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/AssessmentTable/AbstractRpcStatementBulkAction.php:172]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/Document/ElementsService.php [hunk 7]
```diff
@@ -772,12 +772,15 @@ public function prepareElementsForAutoSwitchState(
...
+        string $procedureId,
     ): array {
         $condition = $this->conditionFactory
             ->allConditionsApply(
                 $this->conditionFactory->propertyHasValue($procedureId, $this->elementResourceType->procedure->id),
-                $this->conditionFactory->propertyHasAnyOfValues($elementIdsToSwitch, $this->elementResourceType->id),
+                [] === $elementIdsToSwitch
+                    ? $this->conditionFactory->false()
+                    : $this->conditionFactory->propertyHasAnyOfValues($elementIdsToSwitch, $this->elementResourceType->id),
+                // $this->conditionFactory->propertyHasAnyOfValues($elementIdsToSwitch, $this->elementResourceType->id),
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Document/ElementsService.php:783]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/Elements/PlanningDocumentCategoryTreeReorderer.php [hunk 8]
```diff
@@ -119,10 +122,13 @@ public function getReorderingData(
                 $procedureId,
                 $this->categoryResourceType->procedure->id
             ),
-            $this->conditionFactory->propertyHasAnyOfValues(
+            [] === $categoryToMoveAndNewParentIds
+                ? $this->conditionFactory->false()
+                : $this->conditionFactory->propertyHasAnyOfValues($categoryToMoveAndNewParentIds, $this->categoryResourceType->id),
+            /*$this->conditionFactory->propertyHasAnyOfValues(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Elements/PlanningDocumentCategoryTreeReorderer.php:128]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/News/ProcedureNewsService.php [hunk 9]
```diff
@@ -69,7 +72,10 @@ public function getNewsList($procedureId, $user, $manualSortScope = null, $limit
 
         $roles = $this->determineRoles($roles, $user);
         if (isset($roles) && 0 < count($roles)) {
-            $conditions[] = $this->conditionFactory->propertyHasAnyOfValues($roles, ['roles', 'code']);
+            $conditions[] = [] === $roles
+                ? $this->conditionFactory->false()
+                : $this->conditionFactory->propertyHasAnyOfValues($roles, ['roles', 'code']);
+            // $conditions[] = $this->conditionFactory->propertyHasAnyOfValues($roles, ['roles', 'code']);
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/News/ProcedureNewsService.php:78]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/OwnsProcedureConditionFactory.php [hunk 10]
```diff
@@ -74,7 +76,10 @@ public function isAuthorizedViaPlanningAgency(): FunctionInterface
         $procedure = $this->userOrProcedure;
         $procedurePlanningOffices = $procedure->getPlanningOfficesIds();
 
-        return $this->conditionFactory->propertyHasAnyOfValues($procedurePlanningOffices, ['orga', 'id']);
+        // return $this->conditionFactory->propertyHasAnyOfValues($procedurePlanningOffices, ['orga', 'id']);
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/OwnsProcedureConditionFactory.php:79]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/OwnsProcedureConditionFactory.php [hunk 11]
```diff
@@ -124,7 +131,10 @@ public function hasProcedureAccessingRole(Customer $customer): array
         if (null !== $procedure->getOrgaId()) {
             $this->logger->debug('Permissions: Check whether orga owns procedure');
             $ownsOrgaRoleCondition = [
-                $this->conditionFactory->propertyHasAnyOfValues($relevantRoles, ['roleInCustomers', 'role', 'code']),
+                $conditions[] = [] === $relevantRoles
+                    ? $this->conditionFactory->false()
+                    : $this->conditionFactory->propertyHasAnyOfValues($relevantRoles, ['roleInCustomers', 'role', 'code']),
+                // $this->conditionFactory->propertyHasAnyOfValues($relevantRoles, ['roleInCustomers', 'role', 'code']),
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/OwnsProcedureConditionFactory.php:137]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/OwnsProcedureConditionFactory.php [hunk 12]
```diff
@@ -216,7 +226,12 @@ public function userIsExplicitlyAuthorized(): ClauseFunctionInterface
 
         $procedure = $this->userOrProcedure;
 
-        return $this->conditionFactory->propertyHasAnyOfValues($procedure->getAuthorizedUserIds(), ['id']);
+        $authorizedUserIds = $procedure->getAuthorizedUserIds();
+
+        // return $this->conditionFactory->propertyHasAnyOfValues($procedure->getAuthorizedUserIds(), ['id']);
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/OwnsProcedureConditionFactory.php:231]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/Procedure/ProcedureService.php [hunk 13]
```diff
@@ -2567,7 +2567,10 @@ private function convertFiltersToConditions(
         }
 
         if (isset($filters['municipalCode']) && \is_array($filters['municipalCode'])) {
-            $conditions[] = $this->conditionFactory->propertyHasAnyOfValues($filters['municipalCode'], ['municipalCode']);
+            // $conditions[] = $this->conditionFactory->propertyHasAnyOfValues($filters['municipalCode'], ['municipalCode']);
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Procedure/ProcedureService.php:2570]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/Procedure/ProcedureService.php [hunk 14]
```diff
@@ -2617,8 +2620,14 @@ private function convertFiltersToConditions(
         if (isset($filters['excludeHiddenPhases'])) {
             // Include only procedures where at least one phase is not hidden
             $conditions[] = $this->conditionFactory->anyConditionApplies(
-                $this->conditionFactory->propertyHasNotAnyOfValues($hiddenPhases, ['phase', 'key']),
-                $this->conditionFactory->propertyHasNotAnyOfValues($hiddenPhases, ['publicParticipationPhase', 'key']),
+                // $this->conditionFactory->propertyHasNotAnyOfValues($hiddenPhases, ['phase', 'key']),
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Procedure/ProcedureService.php:2623]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/Procedure/ProcedureService.php [hunk 15]
```diff
@@ -2617,8 +2620,14 @@ private function convertFiltersToConditions(
...
+                [] === $hiddenPhases
+                    ? $this->conditionFactory->false()
+                    : $this->conditionFactory->propertyHasNotAnyOfValues($hiddenPhases, ['phase', 'key']),
+                // $this->conditionFactory->propertyHasNotAnyOfValues($hiddenPhases, ['publicParticipationPhase', 'key']),
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Procedure/ProcedureService.php:2627]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/Logic/Statement/NonAuthorizedAssignRemover.php [hunk 16]
```diff
@@ -115,15 +116,20 @@ private function getAssignableUserIds(Procedure $procedure): array
...
      * @return array<int, Statement|Segment>
+     *
+     * @throws PathException
      */
     private function getClaimablesToUnassign(Procedure $procedure): array
     {
         return $this->statementRepository->getEntities([
             $this->conditionFactory->propertyIsNotNull(['assignee']),
-            $this->conditionFactory->propertyHasNotAnyOfValues(
+            /* $this->conditionFactory->propertyHasNotAnyOfValues(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Statement/NonAuthorizedAssignRemover.php:126]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/ResourceTypes/AssignableUserResourceType.php [hunk 17]
```diff
@@ -63,7 +67,10 @@ protected function getAccessConditions(): array
         }
         if (0 < count($authorizedUsers)) {
             // only return users that are on the list of authorized users
-            return [$this->conditionFactory->propertyHasAnyOfValues($authorizedUserIds, $this->id)];
+            // return [$this->conditionFactory->propertyHasAnyOfValues($authorizedUserIds, $this->id)];
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/AssignableUserResourceType.php:70]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/ResourceTypes/InvitablePublicAgencyResourceType.php [hunk 18]
```diff
@@ -92,10 +92,13 @@ protected function getAccessConditions(): array
                 $this->statusInCustomers->status
             ),
             // avoid already invited organisations
-            $this->conditionFactory->propertyHasNotAnyOfValues(
+            /*$this->conditionFactory->propertyHasNotAnyOfValues(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/InvitablePublicAgencyResourceType.php:95]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/ResourceTypes/PlanningDocumentCategoryResourceType.php [hunk 19]
```diff
@@ -114,7 +116,10 @@ protected function getAccessConditions(): array
         $elementsToHide = $this->globalConfig->getAdminlistElementsHiddenByTitle();
 
         if ([] !== $elementsToHide) {
-            $adminConditions[] = $this->conditionFactory->propertyHasNotAnyOfValues($elementsToHide, $this->title);
+            // $adminConditions[] = $this->conditionFactory->propertyHasNotAnyOfValues($elementsToHide, $this->title);
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/PlanningDocumentCategoryResourceType.php:119]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/ResourceTypes/ReportEntryResourceType.php [hunk 20]
```diff
@@ -77,8 +81,14 @@ protected function getAccessConditions(): array
 
         return [
             $this->conditionFactory->propertyHasValue($procedure->getId(), $this->identifier),
-            $this->conditionFactory->propertyHasAnyOfValues($this->getGroups(), $this->group),
-            $this->conditionFactory->propertyHasAnyOfValues($this->getCategories(), $this->category),
+            // $this->conditionFactory->propertyHasAnyOfValues($this->getGroups(), $this->group),
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/ReportEntryResourceType.php:84]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/ResourceTypes/ReportEntryResourceType.php [hunk 21]
```diff
@@ -77,8 +81,14 @@ protected function getAccessConditions(): array
...
+            [] === $this->getGroups()
+                ? $this->conditionFactory->false()
+                : $this->conditionFactory->propertyHasAnyOfValues($this->getGroups(), $this->group),
+            // $this->conditionFactory->propertyHasAnyOfValues($this->getCategories(), $this->category),
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/ReportEntryResourceType.php:88]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/ResourceTypes/RoleResourceType.php [hunk 22]
```diff
@@ -57,14 +58,21 @@ public function isListAllowed(): bool
...
 
+    /**
+     * @throws PathException
+     */
     protected function getAccessConditions(): array
     {
         $projectRoleCodes = $this->globalConfig->getRolesAllowed();
 
-        return [$this->conditionFactory->propertyHasAnyOfValues(
+        /*return [$this->conditionFactory->propertyHasAnyOfValues(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/RoleResourceType.php:68]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/ResourceTypes/SimilarStatementSubmitterResourceType.php [hunk 23]
```diff
@@ -70,7 +73,11 @@ function (ProcedurePerson $submitter, array $similarStatements): array {
 
                     return [];
                 }),
-            $this->createToOneRelationship($this->procedure)->addConstructorBehavior(new RequiredToOneRelationshipConstructorBehaviorFactory(
+            // FIXME : have to check this change, there is no RequiredToOneRelationshipConstructorBehavior class that's why i used
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/SimilarStatementSubmitterResourceType.php:76]: What about this fixme?

File: demosplan/DemosPlanCoreBundle/ResourceTypes/StatementResourceType.php [hunk 24]
```diff
@@ -127,10 +130,13 @@ public function buildAccessConditions(StatementResourceType $pathStartResourceTy
             $this->conditionFactory->propertyIsNull($pathStartResourceType->headStatement->id),
             // statement placeholders are not considered actual statement resources
             $this->conditionFactory->propertyIsNull($pathStartResourceType->movedStatement),
-            $this->conditionFactory->propertyHasAnyOfValues(
+            /*$this->conditionFactory->propertyHasAnyOfValues(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/StatementResourceType.php:133]: use or remove the comment

File: demosplan/DemosPlanCoreBundle/ResourceTypes/StatementSegmentResourceType.php [hunk 25]
```diff
@@ -103,10 +107,13 @@ protected function getAccessConditions(): array
             ->filterNonOwnedProcedureIds($currentUser, ...$allowedProcedures);
         $procedureIds[] = $procedureId;
 
-        return [$this->conditionFactory->propertyHasAnyOfValues(
+        return [] === $procedureIds
+            ? [$this->conditionFactory->false()]
+            : [$this->conditionFactory->propertyHasAnyOfValues($procedureIds, $this->parentStatementOfSegment->procedure->id)];
+        /*return [$this->conditionFactory->propertyHasAnyOfValues(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/StatementSegmentResourceType.php:113]: use or remove the comment

**Review by mussbach — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/ResourceTypes/SimilarStatementSubmitterResourceType.php (hunk 23 above)
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/SimilarStatementSubmitterResourceType.php]: @mussbach i checked it one more time and it's fixed here https://github.com/demos-europe/demosplan-core/pull/3515/commits/6b95fd57f3d2fdf55ae0fb0d6c4d76794bb3cbd1 . 
The 'RequiredToOneRelationshipConstructorBehavior' does not exists anymore and we have to use now directly the 'RequiredRelationshipConstructorBehavior' which is already adjusted ( see please https://github.com/demos-europe/edt-access-definitions/blob/0.24.42/src/Wrapping/PropertyBehavior/Relationship/ToOne/Factory/RequiredToOneRelationshipConstructorBehaviorFactory.php ).
I tested it too and it worked correctly ( Statement list -> Details and replies -> Edit -> then add or edit Other submitters of the same statement ). I think we use this resource type ( to update and add ) only there

File: composer.json (hunk 1 above)
Comment (bakridemos) [reply composer.json]: this is fixed too, and all unnecessary comments are removed

**Review by mussbach — APPROVED** [review]
let's try it, code review only, thank you for the hard work on this

**Pull request merged**